app/models/work_package.rb

```diff
@@ -64,6 +64,8 @@ class WorkPackage < ApplicationRecord
     order("#{Changeset.table_name}.committed_on ASC, #{Changeset.table_name}.id ASC")
   }
 
+  has_and_belongs_to_many :github_pull_requests
```

Comment:
🤔 `has_and_belongs_to_many` does not have a `dependent: :destroy` option. (at least as far as I can see https://apidock.com/rails/v6.1.3.1/ActiveRecord/Associations/ClassMethods/has_and_belongs_to_many).

Or do you mena re-wiring this `has_and_belongs_to_many` into a `has_many ..., through: ...` connection and then using a `dependent: :destroy` similar to this approach? https://makandracards.com/makandra/32175-don-t-forget-automatically-remove-join-records-on-has_many-through-associations

Reply:
Ah right you are. I also couldn't find that option. In that case one would probably have to do it by hand. Options for that I can think of:
* has_many ..., through: ... as you mentioned
* `after_/before_destroy` hook on work packages
* `on_delete: :cascade` on the db level (but that might be risky in case any postprocessing on GithubPullRequests) is necessary
* Adding it to `WorkPackages::DeleteService`

Of the options mentioned, I'd go for the last one.

Reply:
pairing idea: cron job that deletes un-associated pr's every day (or so) see `app/workers/cron/clear_old_sessions_job.rb` - make sure the class is loaded (in an initializer) so that the cron_job.rb meta programming can pick it up!

Reply:
🏁  done